common/util.go

```diff
@@ -274,12 +328,23 @@ func IsContextCanceledErr(err error) bool {
 		errors.As(err, &canceledSvcErr)
 }
 
-// IsWhitelistServiceTransientError checks if the error is a transient error.
-func IsWhitelistServiceTransientError(err error) bool {
+// IsServiceClientTransientError checks if the error is a transient error.
+func IsServiceClientTransientError(err error) bool {
+	if IsServiceHandlerRetryableError(err) {
+		return true
+	}
+
+	switch err.(type) {
+	case *serviceerror.ResourceExhausted,
+		*serviceerrors.ShardOwnershipLost:
+		return true
+	}
+	return false
+}
+
+func IsServiceHandlerRetryableError(err error) bool {
 	switch err.(type) {
 	case *serviceerror.Internal,
```

Comment:
I am surprised to see `Internal` here. It shouldn't be retryable.

Reply:
I agree. But I am not confident enough that we follow this requirement throughout our codebase so it can be removed from here.  😞